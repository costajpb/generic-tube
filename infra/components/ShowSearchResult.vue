<script setup lang="ts">
    import type Show from 'domain/show/entity'
    import ShowSearchResultGrid from './ShowSearchResultGrid.vue';
    import BaseSkeleton from './BaseSkeleton.vue';

    defineProps<{
        result: Show[]
    }>()
</script>

<template>
    <ShowSearchResultGrid v-show="result && result.length">
        <BaseSkeleton tagName="li" v-for="show in result" :key="show.id">
            <a :class="classes.anchor" :href="'/shows/' + show.id">
                <img :class="classes.cover" :src="show.coverImage" :title="show.title" :alt="show.title" />
            </a>
        </BaseSkeleton>
    </ShowSearchResultGrid>
    <p :class="classes['no-results']" v-show="!result.length">No shows found! :(</p>
</template>

<style module="classes" lang="postcss">
    .anchor {
        position: absolute;
        width: 100%;
        height: 100%;
        text-align: center;
        top: 0;
        left: 0;

        @media (min-width: 700px) {
            position: static;
        }
    }

    .cover {
        max-width: 100%;
        object-fit: cover;
        display: block;
        position: absolute;

        &:not([src]) {
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%);
            /* FIXME */
            color: white;
        }
    }

    .no-results {
        text-align: center;
        font-size: var(--font-size-5);
        font-weight: var(--font-weight-5);
        margin-block: var(--size-4);
        /* FIXME */
        color: var(--gray-6);
    }
</style>