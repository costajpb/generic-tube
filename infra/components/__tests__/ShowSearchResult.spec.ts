import snapshowWrapper from "@/infra/__tests__/snapshotWrapper"
import ShowSearchResult from "../ShowSearchResult.vue"

describe('infra/components/ShowSearchResult', () => {
    it('should render', () => {
        expect(snapshowWrapper(ShowSearchResult, {}, { result: [{
            id: 1,
            title: 'Show title',
            coverImage: 'image'
        }] })).toMatchSnapshot()
    })
})