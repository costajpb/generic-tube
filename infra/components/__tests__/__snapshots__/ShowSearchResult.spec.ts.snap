// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`infra/components/ShowSearchResult > should render 1`] = `
<div
  data-v-app=""
>
  
  <ol
    class="_grid_f639ee"
  >
    
    
    <li
      class="_wrapper_587f33"
    >
      
      <a
        class="_anchor_a1e223"
        href="/shows/1"
      >
        <img
          alt="Show title"
          class="_cover_a1e223"
          src="image"
          title="Show title"
        />
      </a>
      
    </li>
    
    
  </ol>
  <p
    class="_no-results_a1e223"
    style="display: none;"
  >
    No shows found! :(
  </p>
  
</div>
`;
